src/Mapper474.cpp

```diff
@@ -0,0 +1,66 @@
+#include "Mapper474.h"
+#include "Log.h"
+
+namespace sn
+{
+    Mapper474::Mapper474(Cartridge &cart) :
+        Mapper(cart, Mapper::NROM474)
+    {
+        if (cart.getROM().size() == 0xC000) // 3 banks
+        {
+            // this is the only correct ROM size for this mapper
+            m_flavor474 = true;
+        }
+        else
+        {
+            // fail
+            m_flavor474 = false;
+            LOG(Info) << "Incorrectly configured mapper" << std::endl;
+        }
+
+        if (cart.getVROM().size() == 0)
+        {
+            m_usesCharacterRAM = true;
+            m_characterRAM.resize(0x2000);
+            LOG(Info) << "Uses character RAM" << std::endl;
+        }
+        else
+        {
+            m_usesCharacterRAM = false;
+        }
+    }
+
+    Byte MapperNROM::readPRG(Address addr)
```

Comment:
```suggestion
    Byte Mapper474::readPRG(Address addr)
```
Similar for the rest of the functions.